src/MinterGateway.sol

```diff
@@ -794,8 +770,28 @@ contract MinterGateway is IMinterGateway, ContinuousIndexing, ERC712Extended {
 
         if (principalOfMaxAllowedActiveOwedM_ >= principalOfActiveOwedM_) return;
 
+        MinterState storage minterState_ = _minterStates[minter_];
+
+        uint40 penalizeFrom_ = UIntMath.max40(minterState_.updateTimestamp, minterState_.penalizedUntilTimestamp);
+
+        if (newTimestamp_ <= penalizeFrom_) return;
```

Comment:
Effectively here, since `penalizedUntilTimestamp` is set in `_imposePenaltyIfMissedCollateralUpdates`, it means that a Minter can only be penalized for missed collateral update OR undercollateralization but not for missed collateral update AND undercollateralization.
https://github.com/MZero-Labs/protocol/blob/6c8d98526d77a75bd37fd466eec369211da78a36/src/MinterGateway.sol?plain=1#L743

Meaning that a Minter that would stop updating their collateral would only be penalized for missed collateral updates. Which makes sense but is also less harsh than the previous implementation.